controllers/networking-calico/pkg/controller/actuator_reconcile.go

```diff
@@ -0,0 +1,189 @@
+	calicoPSPsSecretName   = "calico-psps"
+
+	noCleanUp = "shoot.gardener.cloud/no-cleanup"
+)
+
+func withYamlSeparator(str string) string {
+	return str + "\n" + "---" + "\n"
+}
+
+func withLocalObjectRefs(refs ...string) []corev1.LocalObjectReference {
+	var localObjectRefs []corev1.LocalObjectReference
+	for _, ref := range refs {
+		localObjectRefs = append(localObjectRefs, corev1.LocalObjectReference{Name: ref})
+	}
+	return localObjectRefs
+}
+
+func Secrets(cl client.Client, calicoFiles *charts.CalicoFiles, namespace string) (*resourcemanager.Secrets, []corev1.LocalObjectReference) {
+	calicoSecret := resourcemanager.NewSecret(cl).
+		WithKeyValues(map[string][]byte{charts.CalicoKey: []byte(calicoFiles.Calico)}).
+		WithNamespacedName(namespace, calicoSecretName)
+
+	calicoConfigSecret := resourcemanager.NewSecret(cl).
+		WithKeyValues(map[string][]byte{charts.CalicoConfigKey: []byte(calicoFiles.Config)}).
+		WithNamespacedName(namespace, calicoConfigSecretName)
+
+	calicoRBAC := resourcemanager.NewSecret(cl).
+		WithKeyValues(map[string][]byte{charts.CalicoRBACKey: []byte(calicoFiles.RBAC)}).
+		WithNamespacedName(namespace, calicoRBACSecretName)
+
+	pspContents := withYamlSeparator(calicoFiles.CalicoTyphaClusterRoleBinding) +
+		withYamlSeparator(calicoFiles.CalicoTyphaClusterRole) +
+		withYamlSeparator(calicoFiles.CalicoClusterRole) +
+		withYamlSeparator(calicoFiles.CalicoTyphaPSP) +
+		withYamlSeparator(calicoFiles.CalicoClusterRoleBinding) +
+		withYamlSeparator(calicoFiles.CalicoPSP) +
+		withYamlSeparator(calicoFiles.CalicoKubeControllersClusterRole) +
+		withYamlSeparator(calicoFiles.CalicoKubeControllersClusterRoleBinding) +
+		calicoFiles.CalicoKubeControllersPSP
+
+	calicoPSPs := resourcemanager.NewSecret(cl).
+		WithKeyValues(map[string][]byte{"psps.yaml": []byte(pspContents)}).
+		WithNamespacedName(namespace, calicoPSPsSecretName)
+
+	return resourcemanager.NewSecrets(cl).WithSecretList([]resourcemanager.Secret{
+		*calicoSecret,
+		*calicoConfigSecret,
+		*calicoRBAC,
+		*calicoPSPs,
+	}), withLocalObjectRefs(calicoSecretName, calicoConfigSecretName, calicoRBACSecretName, calicoPSPsSecretName)
+}
+
+func calicoFileCreated(file string, managedResource *resourcemanagerv1alpha1.ManagedResource) (bool, error) {
+	hasDesired := func(resources []corev1.ObjectReference, desiredObject *unstructured.Unstructured) bool {
+		for _, created := range resources {
+			if created.Name == desiredObject.GetName() &&
+				created.Namespace == desiredObject.GetNamespace() &&
+				created.Kind == desiredObject.GetObjectKind().GroupVersionKind().Kind &&
+				created.APIVersion == desiredObject.GetAPIVersion() {
+				return true
+			}
+		}
+		return false
+	}
+	var (
+		decoder    = yaml.NewYAMLOrJSONDecoder(bytes.NewReader([]byte(file)), 1024)
+		decodedObj map[string]interface{}
+	)
+
+	// A YAML file can contain multiple YAML objects
+	for err := decoder.Decode(&decodedObj); err == nil; err = decoder.Decode(&decodedObj) {
```

Comment:
TBH, I really don't like this in particular. As discussed yesterday we will provide better mechanisms to check the health/status of the managed resources. Having this makes it very hard to update/change anything, it's a lot of complexity, and code that will be deleted for sure anyways very soon. I suggest to remove all of this already now such that we don't need to maintain it.